plugins/kubernetes/app/models/kubernetes/release_doc.rb

```diff
@@ -165,6 +231,49 @@ def disruption_budget_target(deployment)
       end
     end
 
+    def add_env_config_map
+      return unless env_from_config_map?
+      return if env_config_map_exists?
+
+      annotations = reference_annotations.
+        slice(:"samson/override_project_label").
+        merge(
+          "samson/updateTimestamp": Time.now.utc.iso8601,
+          "samson/envConfigMap": true
+        )
+
+      cm = {
+        apiVersion: "v1",
+        kind: "ConfigMap",
+        metadata: {
+          name: env_config_map_name,
+          labels: reference_labels,
+          annotations: annotations
+        },
+        immutable: true,
+        data: static_env
+      }
+
+      if ns = reference_namespace
+        cm[:metadata][:namespace] = ns
+      end
+
+      raw_template << cm
+    end
+
+    def env_config_map_exists?
+      raw_template.any? { |r| r[:kind] == "ConfigMap" && r.dig(:metadata, :annotations, :"samson/envConfigMap") }
+    end
+
+    def env_from_config_map?
+      !!reference_resource&.dig(:metadata, :annotations, :"samson/env_from_config_map")
+    end
+
+    def env_config_map_name
+      version = kubernetes_release.blue_green_color || "blue"
+      "#{reference_name}-#{version}-env"
```

Comment:
```suggestion
      color = kubernetes_release.blue_green_color || "blue"
      "#{reference_name}-#{color}-env"
```
also a little sus that we `||` here since it should always be passed in, so maybe don't pass in the default "blue" to not repeat it here